### PR TITLE
feat: sort project list by computed values and set currency

### PR DESCRIPTION
**Pull request by wreckage0907**
## Description

- Added backend support for sorting the project list view by computed fields (`burn_rate_per_week`, `cost_burn_percent`, `total_budget`, `profit_margin`) using sort-key pagination, ensuring correct ordering and pagination across page boundaries. 
- Introduced helpers to parse `order_by` clauses and compute sort values in Python, mirroring frontend calculations for consistency.
- Updated frontend project list columns and subheader to reflect new computed fields and correct sort field mappings.
- Ensured the `currency` field on `Resource Allocation` is always set from the linked project's `custom_currency` during validation, regardless of incoming payload or user permission resets. 
- Added tests to confirm correct currency assignment in various scenarios, including permission edge cases and explicit overrides. 
- Added integration tests for computed field sorting, verifying correct ordering, pagination, and null handling for all supported computed fields in the project list view.
- Refined cost forecast calculations to exclude past allocations, ensuring only ongoing and future costs are considered in forecasts.
- Minor bugfix in project tracking to ensure forecasted cost to completion is calculated correctly.

## Screenshot/Screencast

image

## Checklist

- [x] I have carefully reviewed the code before submitting it for review.
- [x] This code is adequately covered by unit tests to validate its functionality.
- [x] I have conducted thorough testing to ensure it functions as intended.
- [ ] A member of the QA team has reviewed and tested this PR (To be checked by QA or code reviewer)

Fixes https://github.com/rtCamp/next-pms/issues/1337#issuecomment-4855147615

### CODE REVIEW

File: next_pms/next_projects/api/project.py [hunk 1]
```diff
@@ -100,6 +108,67 @@ def get_profit_margin(total_budget: float, cost_accrued: float, cost_forecasted:
...
+    display), sort with nulls last, and return the page's names plus the total count.
+    """
+    rows = get_list(
+        "Project",
+        fields=SORT_KEY_FIELDS,
+        filters=project_filters,
+        or_filters=or_filters,
+        limit_page_length=0,
+        order_by="modified desc",
+    )
```
[the remark this answers is not in the document]
Comment (wreckage0907) [reply next_pms/next_projects/api/project.py:168]: valid will look into de normalizing the data on a later stage

File: next_pms/next_projects/api/project.py [hunk 2]
```diff
@@ -100,6 +108,67 @@ def get_profit_margin(total_budget: float, cost_accrued: float, cost_forecasted:
...
+        if total_budget <= 0:
+            return 0
+        return (flt(project.get("total_costing_amount")) / total_budget) * 100
+    if sort_field == "total_budget":
+        return total_budget
+    return get_profit_margin(
+        total_budget,
+        flt(project.get("total_costing_amount")),
+        cost_forecasted_map.get(project.get("name"), 0),
+    )
```
[the remark this answers is not in the document]
Comment (wreckage0907) [reply next_pms/next_projects/api/project.py]: fixed

**Review by robinroy03 — APPROVED** [review]

**Review by ayushnirwal — CHANGES_REQUESTED** [review]
Missing functionality - sorting with contract end date. @wreckage0907 does the backend support it?

**Review by ayushnirwal — APPROVED** [review]
